enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,498 @@
+    * Raw value (depending on the value of runtime-kind, the version is read from different files in the container)
+    * Optional - based on the existence of various files in the containers (`$JAVA_HOME/release` for Java, the ELF header of the executable for Golang, the output of `node —version` for Node.js)
+    * Value is a hash of the detected versions: `go1.21.3, v18.18.2, 17.0.2,...` (list not exhaustive)
+  * __`KindImplementer`__ - Entity that provides the runtime-kind implementation 
+    * Raw value (based on the existence of various files in the containers, eg  `$JAVA_HOME/release` for Java)
+    * Optional
+    * Value is a hash of the detected implementers: `Red Hat, inc., Oracle Corporation, Eclipse Adoptium` (list not exhaustive)
+* Runtime Information - The container scanner can selectively identify runtime libraries/frameworks that run in the containers.
+  * __`Name`__ - Name of the runtime used to run the application in the container
+    * Derived value (set by the container scanner)
+    * Optional - based on the capabilities of the container scanner to detect such runtimes
+    * Value is a hash of the detected runtimes `Quarkus, Spring Boot, Apache Tomcat, WildFly, JBoss EAP, RH-SSO, Kafka <x.y>, PostgreSQL` (exhaustive list based on the capabilities of the container scanner, might be enhanced with additional development)
+  * __`Version`__ - The version of the runtime used to run the application
+    * Raw values (read from files in the containers)
+    * Optional - based on the capabilities of the container scanner to detect the runtime and the availability of its version from files in the container
+    * Value is a hash of the detected runtime version: `2.10.3.Final, 7.4.7.GA, 2.7.1, 3.6.0.redhat-00005,...` (list not exhaustive)
+
+It is not planned to add more fields to the `workloadRuntimeInfoContainer` type. However, the container scanner can be enhanced to detect more runtimes without impacting the integration in the Insights Operator.
+
+##### Examples
+
+```json
+"runtimeInfo": {
+  "os": "yqx1WBefoIAq",
+  "osVersion": "AQN-qh_OyRUA",
+  "kind": "wbpgzhNYZQOi",
+  "kindVersion": "cAsoUDy8JFNV",
+  "kindImplementer": "DzP5-GS6tDOQ",
+  "name": "zBVuVhrrC_vC",
+  "version": "lyAE7Oh63WPU"
+}
+```
+
+As the values are derived by the container scanner or correspond to software versions, it is possible to create a dictionary of the hashes and retrieve the unobfuscated values. The example above corresponds to:
+
+```json
+"runtimeInfo": {
+  "os": "rhel",
+  "osVersion": "8.9",
+  "kind": "Java",
+  "kindVersion": "17.0.7",
+  "kindImplementer": "Red Hat, Inc.",
+  "name": "Quarkus",
+  "version": "2.13.8.Final-redhat-00004"
+}
+```
+
+### Workflow Description
+
+In its existing implementation, the OpenShift Insights Operator periodically gathers workload data every 12 hours.
+
+The Operator iterates over all containers in the cluster and captures their Image ID and entry points.
```

Comment:
Not true. There's a current limit of 8000 Pods.

Reply:
I've updated the doc (I mentioned later the 8000 limit but forget about this section)